feathr_project/feathr/registry/_feature_registry_purview.py

```diff
@@ -844,30 +873,55 @@ def _list_registered_entities_with_details(self, project_name: str = None, entit
                 raise RuntimeError(
                     f'only SOURCE, DERIVED_FEATURE, ANCHOR, ANCHOR_FEATURE, FEATHR_PROJECT are supported when listing the registered entities, {entity_type} is not one of them.')
 
-        # the search grammar is less documented in Atlas/Purview. 
-        # Here's the query grammar: https://atlas.apache.org/2.0.0/Search-Advanced.html
-        search_string = "".join(
-            [f" or entityType:{e}" for e in entity_type_list])
-        # remvoe the first additional " or "
-        search_string = search_string[4:]
-        result_entities = self.purview_client.discovery.search_entities(
-            search_string, limit=limit, starting_offset=starting_offset)
+        if project_name is None:
+            raise RuntimeError("You need to specify a project_name")
+        # the search grammar: 
+        # https://docs.microsoft.com/en-us/azure/purview/how-to-search-catalog#search-query-syntax
+        # https://docs.microsoft.com/en-us/rest/api/datacatalog/data-catalog-search-syntax-reference
+
+        # get the corresponding features belongs to a certain project.
+        # note that we need to use "startswith" to filter out the features that don't belong to this project.
+        # see syntax here: https://docs.microsoft.com/en-us/rest/api/purview/catalogdataplane/discovery/query#discovery_query_andornested
+        # this search does the following:
+        # search all the entities that start with project_name+delimiter for all the search entities
+        # However, for TYPEDEF_FEATHR_PROJECT, it doesn't have delimiter in the qualifiedName
+        # Hence if TYPEDEF_FEATHR_PROJECT is in the `entity_type` input, we need to search for that specifically
+        # and finally "OR" the result to union them
+        query_filter = {
```

Comment:
Please format the code, the indentations of the `query_filter` content is a bit of mess.

Reply:
fixec